libkatana_python_native/src/PropertyGraph.cpp

```diff
@@ -363,14 +363,19 @@ DefPropertyGraph(py::module& m) {
              std::optional<std::vector<std::string>> edge_properties,
              TxnContext* txn_ctx) -> std::shared_ptr<PropertyGraph> {
             auto path_str = py::str(path).cast<std::string>();
+            auto res = URI::Make(path_str);
+            if (!res) {
+              KATANA_LOG_FATAL("uri problem {}: {}", path_str, res.error());
```

Comment:
Never use KATANA_LOG_FATAL. You can use `PythonChecked` here. It throws exceptions in the failure case which works well with python.